giraffe/src/components/geo/bing-maps/BingLayerObject.ts

```diff
@@ -0,0 +1,155 @@
+        Math.abs((tilePoint.x + tilePoint.y) % subdomains.length)
+      ]
+    return this._url
+      .replace('{subdomain}', s)
+      .replace('{quadkey}', this.tile2quad(tilePoint.x, tilePoint.y, zoom))
+      .replace('{culture}', this.options.culture)
+  },
+
+  loadMetadata: function() {
+    if (this.metaRequested) return
+    this.metaRequested = true
+    const cbid = '_bing_metadata_' + L.Util.stamp(this)
+    window[cbid] = meta => {
+      window[cbid] = undefined
+      const e = document.getElementById(cbid)
+      e.parentNode.removeChild(e)
+      if (meta.errorDetails) {
+        return
+      }
+      this.initMetadata(meta)
+    }
+    const urlScheme =
+      document.location.protocol === 'file:'
+        ? 'http'
+        : document.location.protocol.slice(0, -1)
+    const url =
+      urlScheme +
+      '://dev.virtualearth.net/REST/v1/Imagery/Metadata/' +
+      this.options.type +
+      '?include=ImageryProviders&jsonp=' +
+      cbid +
+      '&key=' +
+      this._bing_key +
+      '&UriScheme=' +
+      urlScheme +
+      '&culture=' +
+      this.options.culture +
+      '&style=' +
+      this.options.style
+    const script = document.createElement('script')
+    script.type = 'text/javascript'
+    script.src = url
+    script.id = cbid
+    document.getElementsByTagName('head')[0].appendChild(script)
+  },
+
+  initMetadata: function(meta) {
+    const r = meta.resourceSets[0].resources[0]
+    this.options.subdomains = r.imageUrlSubdomains
+    this._url = r.imageUrl
+    if (r.imageryProviders) {
+      for (let i = 0; i < r.imageryProviders.length; i++) {
+        const p = r.imageryProviders[i]
+        for (let j = 0; j < p.coverageAreas.length; j++) {
+          const c = p.coverageAreas[j]
+          const coverage: any = {
+            zoomMin: c.zoomMin,
+            zoomMax: c.zoomMax,
+            active: false,
+          }
+          const bounds = new L.LatLngBounds(
+            new L.LatLng(c.bbox[0] + 0.01, c.bbox[1] + 0.01),
+            new L.LatLng(c.bbox[2] - 0.01, c.bbox[3] - 0.01)
+          )
+          coverage.bounds = bounds
+          coverage.attrib = p.attribution
+          this._providers.push(coverage)
+        }
+      }
+    }
+    this._update()
+  },
+
+  _update: function() {
+    if (this._url === null || !this._map) return
+    this._update_attribution()
+    L.TileLayer.prototype._update.apply(this, [])
+  },
+
+  _update_attribution: function() {
+    const bounds = this._map.getBounds()
+    const zoom = this._map.getZoom()
+    for (let i = 0; i < this._providers.length; i++) {
+      const p = this._providers[i]
+      if (
+        zoom <= p.zoomMax &&
+        zoom >= p.zoomMin &&
+        bounds.intersects(p.bounds)
+      ) {
+        if (!p.active && this._map.attributionControl)
+          this._map.attributionControl.addAttribution(p.attrib)
+        p.active = true
+      } else {
+        if (p.active && this._map.attributionControl)
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.